src/spatialdata/dataloader/graph_loader.py

```diff
@@ -29,104 +46,90 @@ def build_graph(  # TODO: add rings for grids
     method
         The method to use to build the graph. One of "knn", "radius", or "expansion".
     use_centroids
-        Weather to use centroids. Only used when method='expansion'.
+        Weather to use centroids. Assumed use_centroids=True for method='knn' and 'radius'. Only used when
+        method='expansion'.
     k
-        The number of k to use for the knn method.
+        The number of k to use for the method='knn', needs to be None for method='radius' or method='expansion'.
     max_distance
-        The maximum distance between centroids for method='knn' or method='radius', and the buffer for expantion when method='expansion'.
+        The maximum distance between centroids for method='knn' or method='radius', and the buffer for expansion when
+        method='expansion'.
     percentile
-        If max_distance is None, percentile can be used to set max_distance, based on the percentile of pair-wise distances between the centroids.
+        Integer between 0 and 100 (including the endpoints). If max_distance is None, percentile can be used to set
+        max_distance, based on the percentile of pair-wise distances between the centroids. Computing the percentile
+        could take long for large datasets.
     self_loops
-        Weather to keep self loops.
+        Whether to keep self loops.
 
     Returns
     -------
-    The graph in the form of edge indices.
-
+    The graph in the form of edge indices (compatible with PyTorch Geometric).
     """
-    # check method is valid
+    # validate the parameters passed on the method
     assert method in ["knn", "radius", "expansion"]
-    if use_centroids:
-        gdf = gdf.centroid
-    # only one of k, threshold, or max_distance can be specified, the one corresponding to the method selected
+
+    if max_distance is not None:
+        assert max_distance > 0
+    if percentile is not None:
+        assert percentile >= 0
+        assert percentile <= 100
+
     if method == "knn":
         assert k is not None
         assert k > 0
-        assert max_distance is None or max_distance > 0
-        if not use_centroids:
-            logger.warning("method knn is only available for centroids, automatically set use_centroids=True.")
-        gdf_t = gdf.centroid
-        centroids = np.array(gdf_t.apply(lambda g: [g.x, g.y]).tolist())
+        assert not (percentile is not None and max_distance is not None)
+    if method == "radius":
+        assert k is None
+        assert (percentile is None) ^ (max_distance is None)
+    if method == "expansion":
+        assert k is None
+        assert percentile is None
```

Comment:
@EliHei2 I have now moved the validation of the code all above. This makes the code more readable (some cases where missing before, for instance with `method == 'expansion'`, `percentile` was allowed to be not `None`, even if then never used.